### PR TITLE
finishing unit 4 assessment

### PR DESCRIPTION
**Pull request by jayfr1122**
# Submitting Unit 4 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [x] JavaScript Coding Challenges with Jest
- [x] Interview Questions
- [x] Ruby Challenges with RSpec

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -9,28 +9,120 @@
...
+  
+    const fibArray = [0, 1]
+  
+    for (let i = 2; i < length; i++) {
+      const nextFib = fibArray[i - 1] + fibArray[i - 2]
+      fibArray.push(nextFib)
+    }
+  
+    return fibArray
+}
```
Comment (SunkissedQueen) [inline code-challenges.test.js:25]: Good work taking care of edge cases. Consider looking at line 3 condition. The fibonacci needs at least 2 numbers to have a value in index  0 and 1.

File: code-challenges.test.js [hunk 2]
```diff
@@ -9,28 +9,120 @@
...
-const fibonacciLength2 = 10
+const fibLength2 = 10
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
+describe('genFib', () => {
+  it('should generate Fibonacci sequence with length 6', () => {
+    const fibLength1 = 6
+    const result1 = genFib(fibLength1)
+
+    expect(result1).toEqual([0, 1, 1, 2, 3, 5])
```
Comment (SunkissedQueen) [inline code-challenges.test.js:40]: looks like the expected output was changed from what the test variable stated.

File: code-challenges.test.js [hunk 3]
```diff
@@ -9,28 +9,120 @@
...
+  })
+
+  it('should generate Fibonacci sequence with length 10', () => {
+    const fibLength2 = 10;
+    const result2 = genFib(fibLength2);
+
+    expect(result2).toEqual([0, 1, 1, 2, 3, 5, 8, 13, 21, 34])
```
Comment (SunkissedQueen) [inline code-challenges.test.js:47]: looks like the expected output was changed from what the test variable stated.

File: code-challenges.test.js [hunk 4]
```diff
@@ -9,28 +9,120 @@
...
 // Expected output: -31
 
 // b) Create the function that makes the test pass.
+function calcVotes(votes) {
+    if (typeof votes !== 'object' || !votes.hasOwnProperty('upVotes') || !votes.hasOwnProperty('downVotes')) {
+      throw new Error('Input must be an object with upVotes and downVotes properties.')
+    }
+  
+    return votes.upVotes - votes.downVotes
+}
```
Comment (SunkissedQueen) [inline code-challenges.test.js:116]: doing great taking care of edge cases. Please start using ES6 syntax to create the function because they will be needed to pass data in future projects.

**Review by SunkissedQueen — APPROVED** [review]
Jimmy, looks like you were mindful of the edge cases good work. Next assessment, let's start trying to level up from advanced beginner to competent. You seem ready. Try the following:

-  Jest test structure. Instead of creating multiple it methods to test out the edge cases, allow multiple expect statements to perform the function call with each test variable.
- Functions. Use the ES6 so that you become familiar with it before the next set of group projects.
- Pseudocode. Try this structure.
```js
function name:
input:
output:
process:
```
-  Interview Questions: add more content like examples, your experience with the tool, or new info you found during research
Please reach out if you need any help.
